File: core/src/commonMain/kotlin/work/socialhub/kslack/entity/event/ImCloseEvent.java
package work.socialhub.kslack.entity.event;

/**
 * The im_close event is sent to all connections for a user when a direct message channel is closed by that user.
 * <p>
 * https://api.slack.com/events/im_close
 */
public class ImCloseEvent implements Event {

    public static final String TYPE_NAME = "im_close";

    private final String type = TYPE_NAME;
    private String user;
    private String channel;

    public String getType() {
        return this.type;
    }

    public String getUser() {
        return this.user;
    }

    public String getChannel() {
        return this.channel;
    }

    public void setUser(String user) {
        this.user = user;
    }

    public void setChannel(String channel) {
        this.channel = channel;
    }
}

File: core/src/commonMain/kotlin/work/socialhub/kslack/entity/event/ImCloseEvent.kt
package work.socialhub.kslack.entity.event

/**
 * The im_close event is sent to all connections for a user when a direct message channel is closed by that user.
 *
 *
 * https://api.slack.com/events/im_close
 */
class ImCloseEvent : Event {
    override val type: String = TYPE_NAME
    var user: String? = null
    var channel: String? = null

    companion object {
        const val TYPE_NAME: String = "im_close"
    }
}